File: 0234-palindrome-linked-list/0234-palindrome-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        result = []
        n = head
        while(n):
            result.append(n.val)
            n = n.next
        return result == result[::-1]
        

File: 0234-palindrome-linked-list/0234-palindrome-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public boolean isPalindrome(ListNode head) {
        List<Integer> result = new ArrayList<>();
        ListNode n = head;
        while (n != null) {
            result.add(n.val);
            n = n.next;
        }
        
        int start = 0;
        int end = result.size() - 1;
        while (start < end) {
            if (!result.get(start).equals(result.get(end))) {
                return false;
            }
            start++;
            end--;
        }
        
        return true;
        
    }
}
